modules/trimmomatic/main.nf

```diff
@@ -0,0 +1,46 @@
+process TRIMMOMATIC {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::trimmomatic=0.39" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/trimmomatic:0.39--hdfd78af_2':
+        'quay.io/biocontainers/trimmomatic:0.39--hdfd78af_2' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.paired.trim*.fastq.gz")   , emit: trimmed_reads
+    tuple val(meta), path("*.unpaired.trim_*.fastq.gz"), optional:true, emit: unpaired_reads
+    tuple val(meta), path("*.log")                     , emit: log
+    path "versions.yml"                                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def trimmed = meta.single_end ? "SE" : "PE"
+    def output = meta.single_end ?
+        "${prefix}.SE.paired.trim.fastq.gz" // HACK to avoid unpaired and paired in the trimmed_reads output
+        : "${prefix}.paired.trim_1.fastq.gz ${prefix}.unpaired.trim_1.fastq.gz ${prefix}.paired.trim_2.fastq.gz ${prefix}.unpaired.trim_2.fastq.gz"
+    // TODO Give better error output
```

Comment:
Is this something you are leaving for a new iteration or just missed it on this one?

Reply:
Yeah, it works :tm: for now. It's not perfect naming, but trimmomatic is so esoteric with its cli. This is @alyssa-ab's first module and didn't want to spend even more time making it perfect the first go around.

So in conclusion, open to suggestions if anyone has them, but in a new PR.